### PR TITLE
[Merged by Bors] - Add hooks to get the current time and timezone

### PR DESCRIPTION
**Pull request by jedel1043**
This Pull Request changes the following:

- Adds two new hooks to `HostHooks` to access the current UTC time and the current timezone offset.
- Replaces usages of `Local` with the host hook.
- Replaces usages of `Utc::now` and `Local::now` with the hooks.

cc @lastmjs

### CODE REVIEW

File: boa_engine/src/context/hooks.rs
Comment (Razican) [inline boa_engine/src/context/hooks.rs:2]:
```suggestion
```

**Review by Razican — APPROVED** [review]
It looks great to me :) Let's wait for @lastmjs to give some feedback on wether this is good enough for him to use the upstream Boa.

**Comment by lastmjs** [comment]
I will try this out now and let you know

**Comment by lastmjs** [comment]
I can confirm this works, all of our tests have passed.

We're now on upstream Boa with this branch, and will be on main once this is merged :)

Thanks so much, this is just great.

**Review by HalidOdat — APPROVED** [review]
Looks good to me! Lets get this merged! :)

**Pull request closed without merging**
